ffi/derive/src/attr_parse/getset.rs

```diff
@@ -0,0 +1,830 @@
+            }
+        }
+
+        if errors.is_empty() {
+            Ok(SpannedGetSetOptions {
+                span: lit.span(),
+                options: result,
+            })
+        } else {
+            let mut errors = errors.into_iter();
+            let mut error = errors.next().expect("darling::Error can never be empty");
+
+            for next_error in errors {
+                error.combine(next_error);
+            }
+
+            Err(error)
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+pub enum GetSetGenMode {
+    Get,
+    GetCopy,
+    Set,
+    GetMut,
+}
+
+impl Display for GetSetGenMode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            GetSetGenMode::Get => write!(f, "get"),
+            GetSetGenMode::GetCopy => write!(f, "get_copy"),
+            GetSetGenMode::Set => write!(f, "set"),
+            GetSetGenMode::GetMut => write!(f, "get_mut"),
+        }
+    }
+}
+
+impl FromStr for GetSetGenMode {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "get" => Ok(GetSetGenMode::Get),
+            "get_copy" => Ok(GetSetGenMode::GetCopy),
+            "set" => Ok(GetSetGenMode::Set),
+            "get_mut" => Ok(GetSetGenMode::GetMut),
+            _ => Err(()),
+        }
+    }
+}
+
+enum GetSetAttrToken {
+    Skip,
+    Gen(GetSetGenMode, GetSetOptions),
+}
+
+struct SpannedGetSetAttrToken {
+    span: Span,
+    token: GetSetAttrToken,
+}
+
+impl syn2::parse::Parse for SpannedGetSetAttrToken {
+    fn parse(input: ParseStream) -> syn2::Result<Self> {
+        let ident = input.parse::<syn2::Ident>()?;
+
+        match ident.to_string().as_str() {
+            "skip" => Ok(SpannedGetSetAttrToken {
+                span: ident.span(),
+                token: GetSetAttrToken::Skip,
+            }),
+            s @ ("get" | "get_copy" | "set" | "get_mut") => {
+                let mode = s.parse().unwrap();
+
+                if input.peek(Token![=]) {
+                    input.parse::<Token![=]>()?;
+                    let options = input.parse::<SpannedGetSetOptions>()?;
+                    let span = ident
+                        .span()
+                        .join(options.span)
+                        .expect("must be in the same file");
+
+                    Ok(SpannedGetSetAttrToken {
+                        span,
+                        token: GetSetAttrToken::Gen(mode, options.options),
+                    })
+                } else {
+                    Ok(SpannedGetSetAttrToken {
+                        span: ident.span(),
+                        token: GetSetAttrToken::Gen(mode, GetSetOptions::default()),
+                    })
+                }
+            }
+            _ => Err(syn2::Error::new(
+                ident.span(),
+                "expected one of `skip`, `get`, `get_copy`, `set`, `get_mut`",
```

Comment:
This order seems more consistent.

```suggestion
                "expected one of `get`, `get_copy`, `get_mut`, `set`, `skip`",
```